skyportal/handlers/api/news_feed.py

```diff
@@ -61,13 +61,41 @@ def fetch_newest(model):
                 .all()
             )
 
+        def fetch_newest_comments():
+            owned_comments = (
+                Comment.query.filter(
+                    Comment.obj_id.in_(
+                        DBSession()
+                        .query(Source.obj_id)
+                        .filter(
+                            Source.group_id.in_(
+                                [g.id for g in self.current_user.accessible_groups]
+                            )
+                        )
+                    )
+                )
+                .order_by(desc(Comment.created_at or Comment.saved_at))
+                .limit(n_items)
+                .all()
+            )
+
+            for comment in owned_comments:
+                author = User.query.filter(User.username == comment.author).first()
+                comment.author_info = {
```

Comment:
This should be factored out into a function as it's duplicated several times. See comment below in models.py